lib/ansible/modules/cloud/amazon/lightsail.py

```diff
@@ -223,9 +264,44 @@ def create_instance(module, client, instance_name):
             resp = resp['operations'][0]
         except botocore.exceptions.ClientError as e:
             module.fail_json(msg='Unable to create instance {0}, error: {1}'.format(instance_name, e))
-        changed = True
 
-    inst = _find_instance_info(client, instance_name)
+        inst = _find_instance_info(client, instance_name)
+
+        # Wait for instance to become running
+        if wait:
+            while (wait_max > time.time()) and (inst is not None and inst['state']['name'] != "running"):
+                try:
+                    time.sleep(2)
+                    inst = _find_instance_info(client, instance_name)
+                except botocore.exceptions.ClientError as e:
+                    if e.response['ResponseMetadata']['HTTPStatusCode'] == "403":
+                        module.fail_json(msg="Failed to start instance {0}. Check that you have permissions to perform the operation".format(instance_name),
+                                         exception=traceback.format_exc())
+                    elif e.response['Error']['Code'] == "RequestExpired":
+                        module.fail_json(msg="RequestExpired: Failed to start instance {0}.".format(instance_name), exception=traceback.format_exc())
+                    time.sleep(1)
+
+        # Timed out
+        if wait and not changed and wait_max <= time.time():
+            module.fail_json(msg="Wait for instance start timeout at %s" % time.asctime())
+
+        # Attempt to open ports
+        if open_ports:
+            if inst is not None:
+                try:
+                    for o in open_ports:
+                        resp = client.open_instance_public_ports(
+                            instanceName=instance_name,
+                            portInfo={
+                                'fromPort': o['from_port'],
+                                'toPort': o['to_port'],
+                                'protocol': o['protocol']
+                            }
+                        )
+                except botocore.exceptions.ClientError as e:
```

Comment:
botocore.exceptions.BotoCoreError has been added to the AWS guidelines for the exceptions that should be caught. https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/GUIDELINES.md#boto3-1